openstudiocore/src/cli/main.cpp

```diff
@@ -262,249 +279,300 @@ int main(int argc, char *argv[])
+      std::cout << "Unknown Exception in embedded_help" << std::endl; // endl will flush
+      return ruby_cleanup(1);
+    }
+
+    Init_bigdecimal();
+    rb_provide("bigdecimal");
+    rb_provide("bigdecimal.so");
+
+    Init_continuation();
+    rb_provide("continuation");
+    rb_provide("continuation.so");
+
+    Init_coverage();
+    rb_provide("coverage");
+    rb_provide("coverage.so");
+
+    Init_cparse();
+    rb_provide("cparse");
+    rb_provide("cparse.so");
+
+    Init_date_core();
+    rb_provide("date_core");
+    rb_provide("date_core.so");
+
+    Init_digest();
+    rb_provide("digest");
+    rb_provide("digest.so");
+
+    //Init_escape();
+    //rb_provide("escape");
+    //rb_provide("escape.so");
+
+    Init_etc();
+    rb_provide("etc");
+    rb_provide("etc.so");
+
+    Init_fcntl();
+    rb_provide("fcntl");
+    rb_provide("fcntl.so");
+
+    Init_fiber();
+    rb_provide("fiber");
+    rb_provide("fiber.so");
+
+    //Init_fiddle();
+    //rb_provide("fiddle");
+    //rb_provide("fiddle.so");
+
+    Init_generator();
+    rb_provide("generator");
+    rb_provide("generator.so");
+
+    Init_md5();
+    rb_provide("md5");
+    rb_provide("digest/md5");
+    rb_provide("md5.so");
+    rb_provide("digest/md5.so");
+    
+    Init_nkf();
+    rb_provide("nkf");
+    rb_provide("nkf.so");
+
+    Init_nonblock();
+    rb_provide("nonblock");
+    rb_provide("nonblock.so");
+
+    Init_objspace();
+    rb_provide("objspace");
+    rb_provide("objspace.so");
+
+    Init_parser();
+    rb_provide("parser");
+    rb_provide("parser.so");
+
+    Init_pathname();
+    rb_provide("pathname");
+    rb_provide("pathname.so");
+
+    Init_psych();
+    rb_provide("psych");
+    rb_provide("psych.so");
+
+    Init_ripper();
+    rb_provide("ripper");
+    rb_provide("ripper.so");
+
+    Init_rmd160();
+    rb_provide("rmd160");
+    rb_provide("digest/rmd160");
+    rb_provide("rmd160.so");
+    rb_provide("digest/rmd160.so");
+    
+    Init_sdbm();
+    rb_provide("sdbm");
+    rb_provide("sdbm.so");
+
+    Init_sha1();
+    rb_provide("sha1");
+    rb_provide("digest/sha1");
+    rb_provide("sha1.so");
+    rb_provide("digest/sha1.so");
+    
+    Init_sha2();
+    rb_provide("sha2");
+    rb_provide("digest/sha2");
+    rb_provide("sha2.so");
+    rb_provide("digest/sha2.so");
+    
+    Init_sizeof();
+    rb_provide("sizeof");
+    rb_provide("sizeof.so");
+
+    Init_socket();
+    rb_provide("socket");
+    rb_provide("socket.so");
+
+    Init_stringio();
+    rb_provide("stringio");
+    rb_provide("stringio.so");
+
+    Init_strscan();
+    rb_provide("strscan");
+    rb_provide("strscan.so");
+
+    Init_thread();
+    rb_provide("thread");
+    rb_provide("thread.so");
+
+    #ifndef _MSC_VER
+      Init_wait();
+      rb_provide("wait");
+      rb_provide("wait.so");
+    #endif
+
+    Init_zlib();
+    rb_provide("zlib");
+    rb_provide("zlib.so");
+
+    #ifndef __APPLE__
```

Comment:
@tijcolem I am not sure why OpenSSL is disabled on Apple.  If the custom version of Ruby can be build with OpenSSL support then I think this can be enabled on all platforms

@kbenne FYI